### PR TITLE
create DebugBar and view of topic and controller Topic

### PR DESCRIPTION
**Pull request by tienquanptit**

### CODE REVIEW

File: app/Http/Controllers/admin/TopicController.php [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        $this->topic = $topic;
+    }
+
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
```
Comment (framgiaciv3) [inline app/Http/Controllers/admin/TopicController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/admin/TopicController.php</br>**Line**: 35</br>Variable "test_model" is not in valid camel caps format

File: app/Http/Controllers/admin/TopicController.php [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+        $topics = $this->topic->getAll();
```
Comment (framgiaciv3) [inline app/Http/Controllers/admin/TopicController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/admin/TopicController.php</br>**Line**: 36</br>Variable "fillable_columns" is not in valid camel caps format

Comment (framgiaciv3) [inline app/Http/Controllers/admin/TopicController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/admin/TopicController.php</br>**Line**: 36</br>Variable "test_model" is not in valid camel caps format

File: app/Http/Controllers/admin/TopicController.php [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+        $test_model = new Topic();
```
Comment (framgiaciv3) [inline app/Http/Controllers/admin/TopicController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/admin/TopicController.php</br>**Line**: 37</br>Variable "fillable_columns" is not in valid camel caps format

File: app/Http/Controllers/admin/TopicController.php [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+        $fillable_columns = $test_model->getFillable();
```
Comment (framgiaciv3) [inline app/Http/Controllers/admin/TopicController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/admin/TopicController.php</br>**Line**: 38</br>Variable "test_columns" is not in valid camel caps format

File: app/Http/Controllers/admin/TopicController.php [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(TopicFormRequest $request)
+    {
```
Comment (framgiaciv3) [inline app/Http/Controllers/admin/TopicController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/admin/TopicController.php</br>**Line**: 63</br>Variable "user_id" is not in valid camel caps format

File: app/Http/Controllers/admin/TopicController.php [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+        if (Auth::check()) {
```
Comment (framgiaciv3) [inline app/Http/Controllers/admin/TopicController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/admin/TopicController.php</br>**Line**: 64</br>Variable "user_id" is not in valid camel caps format

File: app/Http/Controllers/admin/TopicController.php [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+            $user_id = Auth::user()->id;
+            $request->merge(['user_id' => $user_id]);
+            $image = $request->file('image_topic');
+            $img = $this->saveImage($image);
+            $request->merge(['images' => $img]);
+
+            //handle select time
```
Comment (framgiaciv3) [inline app/Http/Controllers/admin/TopicController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/admin/TopicController.php</br>**Line**: 71</br>Variable "select_time" is not in valid camel caps format

File: app/Http/Controllers/admin/TopicController.php [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+            if ($request->input('set_time') == 0) {
+                $select_time = null;
```
Comment (framgiaciv3) [inline app/Http/Controllers/admin/TopicController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/admin/TopicController.php</br>**Line**: 73</br>Variable "select_time" is not in valid camel caps format

File: app/Http/Controllers/admin/TopicController.php [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+            } elseif ($request->input('set_time') == 1) {
```
Comment (framgiaciv3) [inline app/Http/Controllers/admin/TopicController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.ControlStructureSpacing.SpacingBeforeClose</br>**Severity**: error</br>**File**: app/Http/Controllers/admin/TopicController.php</br>**Line**: 74</br>Blank line found at end of control structure

File: app/Http/Controllers/admin/TopicController.php [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+                $select_time = $request->input('select_time');
+
```
Comment (framgiaciv3) [inline app/Http/Controllers/admin/TopicController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/admin/TopicController.php</br>**Line**: 76</br>Variable "select_time" is not in valid camel caps format

File: app/Http/Controllers/admin/TopicController.php [hunk 11]
```diff
@@ -0,0 +1,187 @@
...
+            }
+            $request->merge(['select_time' => $select_time]);
+//            dd(Carbon::now()->hour . ':'. Carbon::now()->minute);
+            $this->topic->store($request->all());
+
+            return back()->with('success', __('message.success'));
+        }
```
Comment (framgiaciv3) [inline app/Http/Controllers/admin/TopicController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionClosingBrace.SpacingBeforeClose</br>**Severity**: error</br>**File**: app/Http/Controllers/admin/TopicController.php</br>**Line**: 83</br>Function closing brace must go on the next line following the body; found 1 blank lines before brace

File: app/Http/Controllers/admin/TopicController.php [hunk 12]
```diff
@@ -0,0 +1,187 @@
...
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param  \Illuminate\Http\Request $request
+     * @param  int $id
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request, $id)
+    {
```
Comment (framgiaciv3) [inline app/Http/Controllers/admin/TopicController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/admin/TopicController.php</br>**Line**: 132</br>Variable "column_name" is not in valid camel caps format

File: app/Http/Controllers/admin/TopicController.php [hunk 13]
```diff
@@ -0,0 +1,187 @@
...
+        $test = Topic::find($id);
```
Comment (framgiaciv3) [inline app/Http/Controllers/admin/TopicController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/admin/TopicController.php</br>**Line**: 133</br>Variable "column_value" is not in valid camel caps format

File: app/Http/Controllers/admin/TopicController.php [hunk 14]
```diff
@@ -0,0 +1,187 @@
...
+        $column_name = Input::get('name');
+        $column_value = Input::get('value');
+        if (Input::has('name')) {
+            $test = Topic::select()
```
Comment (framgiaciv3) [inline app/Http/Controllers/admin/TopicController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/admin/TopicController.php</br>**Line**: 137</br>Variable "column_name" is not in valid camel caps format

Comment (framgiaciv3) [inline app/Http/Controllers/admin/TopicController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/admin/TopicController.php</br>**Line**: 137</br>Variable "column_value" is not in valid camel caps format

File: app/Http/Controllers/admin/TopicController.php [hunk 15]
```diff
@@ -0,0 +1,187 @@
...
+                ->where('id', '=', $id)
+                ->update([$column_name => $column_value]);
+
+            return response()->json(['code' => 200], 200);
+        }
+
+        return response()->json(['error' => 400, 'message' => 'Not enought params'], 400);
+    }
```
Comment (framgiaciv3) [inline app/Http/Controllers/admin/TopicController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR1.Methods.CamelCapsMethodName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/admin/TopicController.php</br>**Line**: 145</br>Method name "TopicController::bulk_update" is not in camel caps format

File: app/Http/Controllers/admin/TopicController.php [hunk 16]
```diff
@@ -0,0 +1,187 @@
...
+
+    public function bulk_update(Request $request)
+    {
+        if (Input::has('ids_to_edit') && Input::has('bulk_name') && Input::has('bulk_value')) {
```
Comment (framgiaciv3) [inline app/Http/Controllers/admin/TopicController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/admin/TopicController.php</br>**Line**: 149</br>Variable "bulk_name" is not in valid camel caps format

File: app/Http/Controllers/admin/TopicController.php [hunk 17]
```diff
@@ -0,0 +1,187 @@
...
+            $ids = Input::get('ids_to_edit');
```
Comment (framgiaciv3) [inline app/Http/Controllers/admin/TopicController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/admin/TopicController.php</br>**Line**: 150</br>Variable "bulk_value" is not in valid camel caps format

File: app/Http/Controllers/admin/TopicController.php [hunk 18]
```diff
@@ -0,0 +1,187 @@
...
+            $bulk_name = Input::get('bulk_name');
+            $bulk_value = Input::get('bulk_value');
```
Comment (framgiaciv3) [inline app/Http/Controllers/admin/TopicController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/admin/TopicController.php</br>**Line**: 152</br>Variable "bulk_name" is not in valid camel caps format

Comment (framgiaciv3) [inline app/Http/Controllers/admin/TopicController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/admin/TopicController.php</br>**Line**: 152</br>Variable "bulk_value" is not in valid camel caps format

File: app/Http/Controllers/admin/TopicController.php [hunk 19]
```diff
@@ -0,0 +1,187 @@
...
+            foreach ($ids as $id) {
+                $test = Test::select()->where('id', '=', $id)->update([$bulk_name => $bulk_value]);
```
Comment (framgiaciv3) [inline app/Http/Controllers/admin/TopicController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Controllers/admin/TopicController.php</br>**Line**: 154</br>String "Done" does not require double quotes; use single quotes instead

File: app/Http/Controllers/admin/TopicController.php [hunk 20]
```diff
@@ -0,0 +1,187 @@
...
+            }
+            $message = "Done";
```
Comment (framgiaciv3) [inline app/Http/Controllers/admin/TopicController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Controllers/admin/TopicController.php</br>**Line**: 156</br>String "Error. Empty or Wrong data provided." does not require double quotes; use single quotes instead

File: app/Http/Controllers/admin/TopicController.php [hunk 21]
```diff
@@ -0,0 +1,187 @@
...
+        } else {
+            $message = "Error. Empty or Wrong data provided.";
```
Comment (framgiaciv3) [inline app/Http/Controllers/admin/TopicController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.ShortArraySyntax</br>**Severity**: error</br>**File**: app/Http/Controllers/admin/TopicController.php</br>**Line**: 158</br>"array()" function used. Use short array syntax, "[]", instead.

File: app/Http/Controllers/admin/TopicController.php [hunk 22]
```diff
@@ -0,0 +1,188 @@
...
+        if (Input::has('name')) {
+            $test = Topic::select()
+                ->where('id', '=', $id)
+                ->update([$columnName => $columnValue]);
+
+            return response()->json(['code' => 200], 200);
+        }
+
+        return response()->json(['error' => 400, 'message' => trans('message.not_enought_params')], 400);
+    }
```
Comment (framgiaciv3) [inline app/Http/Controllers/admin/TopicController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR1.Methods.CamelCapsMethodName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/admin/TopicController.php</br>**Line**: 143</br>Method name "TopicController::bulk_update" is not in camel caps format

File: app/Http/Controllers/admin/TopicController.php [hunk 23]
```diff
@@ -0,0 +1,188 @@
...
+        return back()->with('sucess', __('message.sucess'));
+    }
+
+    public function saveImage($image)
+    {
+//        dd($image);
+//        save in storage
+        $filenameWithExt = $image->getClientOriginalName();
+        $filename = pathinfo($filenameWithExt, PATHINFO_FILENAME);
+        $extension = $image->getClientOriginalExtension();
```
Comment (framgiaciv3) [inline app/Http/Controllers/admin/TopicController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/admin/TopicController.php</br>**Line**: 183</br>Variable "new_name" is not in valid camel caps format

File: app/Http/Controllers/admin/TopicController.php [hunk 24]
```diff
@@ -0,0 +1,188 @@
...
+
```
Comment (framgiaciv3) [inline app/Http/Controllers/admin/TopicController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/admin/TopicController.php</br>**Line**: 184</br>Variable "new_name" is not in valid camel caps format

File: app/Http/Controllers/admin/TopicController.php [hunk 25]
```diff
@@ -0,0 +1,188 @@
...
+        $new_name = $filename . '_' . time() . '.' . $extension;
+        $path = $image->storeAs('images/topics', $new_name);
```
Comment (framgiaciv3) [inline app/Http/Controllers/admin/TopicController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/admin/TopicController.php</br>**Line**: 186</br>Variable "new_name" is not in valid camel caps format

File: app/Http/Controllers/admin/TopicController.php [hunk 26]
```diff
@@ -0,0 +1,188 @@
...
+            $request->merge(['images' => $img]);
+
+            //handle select time
+            if ($request->input('set_time') == 0) {
+                $selectTime = null;
+            } elseif ($request->input('set_time') == 1) {
+                $selectTime = $request->input('select_time');
+            }
+            $request->merge(['select_time' => $selectTime]);
+//            dd(Carbon::now()->hour . ':'. Carbon::now()->minute);
```
Comment (BunnyPi04) [inline app/Http/Controllers/admin/TopicController.php]: xóa đi nhé

File: app/Http/Controllers/admin/TopicController.php [hunk 27]
```diff
@@ -0,0 +1,188 @@
...
+    public function destroy($id)
+    {
+        $this->topic->delete($id);
+
+        return back()->with('sucess', __('message.sucess'));
+    }
+
+    public function saveImage($image)
+    {
+//        dd($image);
```
Comment (BunnyPi04) [inline app/Http/Controllers/admin/TopicController.php]: xóa đi

File: resources/js/upload_file.js [hunk 28]
```diff
@@ -0,0 +1,143 @@
...
+                        // document.location.reload(true);
+                    }
+                };
+
+                // Start upload
+                // xhr.open('POST', document.getElementById('file-upload-form').action, true);
+                // xhr.setRequestHeader('X-File-Name', file.name);
+                // xhr.setRequestHeader('X-File-Size', file.size);
+                // xhr.setRequestHeader('Content-Type', 'multipart/form-data');
+                // xhr.send(file);
```
Comment (BunnyPi04) [inline resources/js/upload_file.js]: không dùng xóa đi

File: resources/js/upload_file.js [hunk 29]
```diff
@@ -0,0 +1,143 @@
...
+
+    // Check for the various File API support.
+    if (window.File && window.FileList && window.FileReader) {
+        Init();
+    } else {
+        document.getElementById('file-drag').style.display = 'none';
+    }
+}
+
+ekUpload();
```
Comment (BunnyPi04) [inline resources/js/upload_file.js]: thiếu dòng trắng cuối file

File: resources/views/admin/topic/create.blade.php [hunk 30]
```diff
@@ -0,0 +1,204 @@
...
+                    <div class="form-group form-md-line-input {{ $errors->has('image_topic') ? 'has-error' : '' }}">
+                        <label class="col-md-3 control-label"
+                               for="form_control_1">{{ trans('message.label_image') }}</label>
+                        <div class="col-md-9">
+                            <!-- Upload  -->
+                            <div id="file-upload-form" class="uploader">
+                                {!! Form::file('image_topic', ['id' => 'file-upload', 'accept' => 'image/*', 'enctype' => 'multipart/form-data']) !!}
+                                <span class="text-danger">{{ $errors->first('image_topic') }}</span>
+                                <label for="file-upload" id="file-drag">
+                                    <img id="file-image" src="#" alt="Preview" class="hidden">
```
Comment (BunnyPi04) [inline resources/views/admin/topic/create.blade.php]: sử dụng collective

File: resources/views/admin/topic/create.blade.php [hunk 31]
```diff
@@ -0,0 +1,204 @@
...
+        });
+
+        function hidden_select_time() {
+            document.getElementById('show_select_time_cacular').style.display = 'none';
+            console.log('none');
+        }
+
+        function select_time_cacular() {
+            document.getElementById('show_select_time_cacular').style.display = 'block';
+            console.log('block');
```
Comment (BunnyPi04) [inline resources/views/admin/topic/create.blade.php]: xóa console log

File: resources/views/admin/topic/edit.blade.php [hunk 32]
```diff
@@ -0,0 +1,201 @@
...
+            document.getElementById('show_select_time_cacular').style.display = 'none';
+            console.log('none');
+        }
+
+        function select_time_cacular() {
+            document.getElementById('show_select_time_cacular').style.display = 'block';
+            console.log('block');
+        }
+    </script>
+@endsection
```
Comment (BunnyPi04) [inline resources/views/admin/topic/edit.blade.php]: thiếu dòng trắng cuối file

File: resources/views/admin/topic/index.blade.php [hunk 33]
```diff
@@ -0,0 +1,157 @@
...
+                                                    class="btn btn-xs btn-circle green-jungle ladda-label">{{ trans('message.active') }}</label>
+                                        @else
+                                            <label
+                                                    class="btn btn-xs btn-circle grey-mint">{{ trans('message.deactivate') }}</label>
+                                        @endif
+                                    </td>
+                                    <td>
+                                        @if ($topic->set_time == 1)
+                                            <label
+                                                    class="btn btn-xs btn-circle green-sharp">{{ trans('message.timer') }}</label>
```
Comment (BunnyPi04) [inline resources/views/admin/topic/index.blade.php]: tại sao đoạn này lại phải xuống dòng?

File: resources/views/admin/topic/index.blade.php [hunk 34]
```diff
@@ -0,0 +1,157 @@
...
+                        return 'Server error. Check entered data.';
+                    } else {
+
+                        return response.responseText;
+                    }
+                }
+            });
+        });
+    </script>
+@endsection
```
Comment (BunnyPi04) [inline resources/views/admin/topic/index.blade.php]: thiếu dòng trắng cuối file

File: resources/views/admin/topic/index.blade.php [hunk 35]
```diff
@@ -0,0 +1,157 @@
...
+    <script src="{{ asset('bower_components/demo-bower/confession/admin/assets/global/plugins/bootstrap-editable/bootstrap-editable/js/bootstrap-editable.js') }}"
+            type="text/javascript"></script>
+
+    <script>
+        // $.fn.editable.defaults.mode = 'inline';
+        $(document).ready(function () {
+            $('.testEdit').editable({
+                params: function (params) {
+                    // add additional params from data-attributes of trigger element
+                    params._token = $("#_token").data("token");
```
Comment (BunnyPi04) [inline resources/views/admin/topic/index.blade.php]: sử dụng nháy đơn

File: resources/views/admin/topic/index.blade.php [hunk 36]
```diff
@@ -0,0 +1,157 @@
...
+                    params.name = $(this).editable().data('name');
+
+                    return params;
+                },
+                error: function (response, newValue) {
+                    if (response.status === 500) {
+
+                        return 'Server error. Check entered data.';
```
Comment (BunnyPi04) [inline resources/views/admin/topic/index.blade.php]: i18n

File: resources/views/admin/topic/show.blade.php [hunk 37]
```diff
@@ -0,0 +1,167 @@
...
+ */
+?>
+<?php
+/**
+ * Created by PhpStorm.
+ * User: quantien
+ * Date: 11/1/18
+ * Time: 6:11 PM
+ */
+?>
```
Comment (BunnyPi04) [inline resources/views/admin/topic/show.blade.php]: xóa đi

File: resources/views/admin/topic/show.blade.php [hunk 38]
```diff
@@ -0,0 +1,167 @@
...
+                        {{--delete--}}
+                        {!! Form::open(['method' => 'DELETE', 'class' => 'display_form', 'route' => ['topics.destroy', 'id' => $topic->id]]) !!}
+                        {!! Form::button('<i class="fa fa-trash-o"></i> ' . trans('message.btn_delete'), ['class' => 'btn btn-outline btn-circle btn-sm red-thunderbird', 'type' => 'submit']) !!}
+                        {!! Form::close() !!}
+                    </div>
+                </div>
+                <div class="row">
+                    <div class="col-md-8">
+                          @php
+                              $url = (isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] === 'on' ? "https" : "http") . "://$_SERVER[HTTP_HOST]";
```
Comment (BunnyPi04) [inline resources/views/admin/topic/show.blade.php]: sử dụng nháy đơn

File: app/Http/Controllers/admin/TopicController.php [hunk 39]
```diff
@@ -0,0 +1,186 @@
...
+
+    /**
+     * Show the form for editing the specified resource.
+     *
+     * @param  int $id
+     * @return \Illuminate\Http\Response
+     */
+    public function edit($id)
+    {
+        $topic = $this->topic->find($id);
```
Comment (BunnyPi04) [inline app/Http/Controllers/admin/TopicController.php]: findOrFail

File: app/Http/Controllers/admin/TopicController.php [hunk 40]
```diff
@@ -0,0 +1,186 @@
...
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param  \Illuminate\Http\Request $request
+     * @param  int $id
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request, $id)
+    {
+        $test = Topic::find($id);
```
Comment (BunnyPi04) [inline app/Http/Controllers/admin/TopicController.php]: đây nữa

File: resources/js/admin.js [hunk 41]
```diff
@@ -0,0 +1,30 @@
+function ChangeToSlug() {
+    var title, slug;
+    //Lấy text từ thẻ input title
+    title = document.getElementById("title").value;
```
Comment (BunnyPi04) [inline resources/js/admin.js]: sử dụng nháy đơn

File: resources/views/admin/topic/show.blade.php [hunk 42]
```diff
@@ -0,0 +1,152 @@
...
+                            </div>
+                        </div>
+                        <!-- END CONDENSED TABLE PORTLET-->
+                    </div>
+                </div>
+            </div>
+            <!-- END SAMPLE TABLE PORTLET-->
+        </div>
+    </div>
+@endsection
```
Comment (BunnyPi04) [inline resources/views/admin/topic/show.blade.php]: thiếu dòng trắng cuối file

File: resources/views/admin/topic/show.blade.php [hunk 43]
```diff
@@ -0,0 +1,152 @@
...
+                                                            class="btn btn-xs btn-circle grey-mint">{{ trans('message.deactivate') }}</label>
+                                                @endif
+                                            </td>
+                                        </tr>
+                                        <tr>
+                                            <th>{{ trans('message.set_time') }}</th>
+                                            <td>
+                                                @if ($topic->set_time == 1)
+                                                    <label
+                                                            class="btn btn-xs btn-circle blue-madison">{{ trans('message.timer') }}</label>
```
Comment (BunnyPi04) [inline resources/views/admin/topic/show.blade.php]: sao chỗ này lại xuống dòng?

File: resources/js/upload_file.js [hunk 44]
```diff
@@ -0,0 +1,136 @@
+// File Upload
+//
+function ekUpload() {
+    function Init() {
+        console.log("Upload Initialised");
```
Comment (BunnyPi04) [inline resources/js/upload_file.js]: xóa console.log

File: resources/js/upload_file.js [hunk 45]
```diff
@@ -0,0 +1,136 @@
...
+            document.getElementById('file-image').src = URL.createObjectURL(file);
+            document.getElementById("file-image-input").setAttribute('value', file.name);
+            // console.log(URL.createObjectURL(file));
+        }
+        else {
+            document.getElementById('file-image').classList.add("hidden");
+            document.getElementById('notimage').classList.remove("hidden");
+            document.getElementById('start').classList.remove("hidden");
+            document.getElementById('response').classList.add("hidden");
+            document.getElementById("file-upload-form").reset();
```
Comment (BunnyPi04) [inline resources/js/upload_file.js]: sử dụng nháy đơn hết

File: resources/views/admin/topic/create.blade.php [hunk 46]
```diff
@@ -0,0 +1,202 @@
...
+                            <div class="radio">
+                                <label class="mt-radio mt-radio-outline padding-left">
+                                    {{ Form::radio('set_time', '1' , true, ['id' => 'radClickChecked', 'onclick' => 'select_time_cacular()']) }}
+                                    {{ trans('message.timer') }}
+                                    <span></span>
+                                </label>
+                            </div>
+                            <div class="radio">
+                                <label class="mt-radio mt-radio-outline padding-left">
+                                    {{ Form::radio('set_time', '0' , false, ['id' => 'radClickChecked', 'onclick' => 'hidden_select_time()']) }}
```
Comment (BunnyPi04) [inline resources/views/admin/topic/create.blade.php]: convention

File: resources/views/admin/topic/edit.blade.php [hunk 47]
```diff
@@ -0,0 +1,201 @@
...
+                            <div class="radio">
+                                <label class="mt-radio mt-radio-outline padding-left">
+                                    {{ Form::radio('status', '1' , true, ['id' => 'radClickChecked']) }}
+                                    {{ trans('message.active') }}
+                                    <span></span>
+                                </label>
+                            </div>
+                            <div class="radio">
+                                <label class="mt-radio mt-radio-outline padding-left">
+                                    {{ Form::radio('status', '0' , false, ['id' => 'radClickChecked']) }}
```
Comment (BunnyPi04) [inline resources/views/admin/topic/edit.blade.php]: convention

File: resources/views/admin/topic/edit.blade.php [hunk 48]
```diff
@@ -0,0 +1,201 @@
...
+                    </div>
+
+                    {{--status--}}
+                    <div class="form-group form-md-line-input {{ $errors->has('status') ? 'has-error' : '' }}">
+                        <label class="col-md-3 control-label"
+                               for="form_control_1">{{ trans('message.label_status') }}</label>
+                        <div class="col-md-9">
+                            <div class="radio">
+                                <label class="mt-radio mt-radio-outline padding-left">
+                                    {{ Form::radio('status', '1' , true, ['id' => 'radClickChecked']) }}
```
Comment (BunnyPi04) [inline resources/views/admin/topic/edit.blade.php]: convention

File: resources/views/admin/topic/edit.blade.php [hunk 49]
```diff
@@ -0,0 +1,201 @@
...
+        });
+
+        function hidden_select_time() {
+            document.getElementById('show_select_time_cacular').style.display = 'none';
+            console.log('none');
+        }
+
+        function select_time_cacular() {
+            document.getElementById('show_select_time_cacular').style.display = 'block';
+            console.log('block');
```
Comment (BunnyPi04) [inline resources/views/admin/topic/edit.blade.php]: xóa console.log

File: resources/views/admin/topic/index.blade.php [hunk 50]
```diff
@@ -0,0 +1,161 @@
...
+                </div>
+                <div class="portlet-body">
+                    <div class="actions">
+                        {{--create--}}
+                        <a href="{{ route('topics.create') }}"
+                           class="btn blue mt-ladda-btn ladda-button btn-circle">
+                            {{ trans('message.btn_create') }} </a>
+                    </div>
+                    <div class="table-scrollable">
+                        {!! Form::open(['route' => 'topics/bulk_update', 'method' => "POST", "class"=>"form-inline"]) !!}
```
Comment (BunnyPi04) [inline resources/views/admin/topic/index.blade.php]: convention
sử dụng nháy đơn

File: webpack.mix.js [hunk 51]
```diff
@@ -14,3 +14,5 @@ const mix = require('laravel-mix');
 mix.js('resources/js/app.js', 'public/js')
    .sass('resources/sass/app.scss', 'public/css');
 mix.js('resources/js/user.js', 'public/js');
+mix.copyDirectory('resources/js/admin.js', 'public/js');
+mix.copyDirectory('resources/js/upload_file.js', 'public/js');
```
Comment (BunnyPi04) [inline webpack.mix.js]: thiếu dòng trắng cuối file

**Pull request merged**